### PR TITLE
Feature/mmw 45 jwt tokens

### PR DESCRIPTION
**Pull request by kdelchev**
Put JWT code into a gem.

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -0,0 +1 @@
+ruby-2.3.0
```
Comment (austinmoore) [inline .ruby-version]: Use the oldest ruby version that we use. Probably the version that we use in core. That way we can ensure that we are backwards compatible.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,85 @@
...
+# Shore::Client
+
+Shore client to authorize API calls using [JWT](https://jwt.io)
+
+## Installation
+
+Add this line to your application's Gemfile:
+
+```ruby
+gem 'shore-client', github: 'shore-gmbh/shore-ruby-client', tag: 'v0.1.0'
```
Comment (austinmoore) [inline README.md:10]: Perfect.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,85 @@
...
+Or install it yourself as:
+
+```sh
+$ gem install shore-client
+```
+
+## Usage
+
+```ruby
+require 'shore/client'
```
Comment (austinmoore) [inline README.md]: Is it 'shore/client' or 'shore-client'? If we could get it to be 'shore-client', then bundle will correctly require it. That would be nice. I always find it so annoying when gems have a different require than their name.

File: lib/shore/client/tokens/access_token.rb [hunk 4]
```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
```
Comment (austinmoore) [inline lib/shore/client/tokens/access_token.rb]: If we have use ruby <2.3, then we can't use this nice stuff. :(

File: lib/shore/client/tokens/merchant_account.rb [hunk 5]
```diff
@@ -0,0 +1,111 @@
+# frozen_string_literal: true
+module Shore
+  module Client
+    module Tokens
+      class MerchantAccount # :nodoc:
+        require_relative 'merchant_role'
```
Comment (austinmoore) [inline lib/shore/client/tokens/merchant_account.rb]: Move the requires to the top of the file. That's the ruby convention, I think. Right?

File: spec/shore/client/tokens/access_token_spec.rb [hunk 6]
```diff
@@ -0,0 +1,132 @@
+require 'spec_helper'
+
+describe Shore::Client::Tokens::AccessToken do
```
Comment (austinmoore) [inline spec/shore/client/tokens/access_token_spec.rb]: Let's use the strictest possible RSpec syntax. With that the outer `describe ...` must be `RSpec.describe ...`. All of the other calls to describe within the block don't need to be prefixed with `RSpec.` though. Take a look at the spec_helper.rb in the messaging service. I think this has all of the strictest settings. I copied it from Core. Core we upgraded from RSpec 2 to 3 and the strict settings made it easier to upgrade.

File: spec/shore/client/tokens/merchant_account_spec.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+            'slug' => 'achsel-alex',
+            'name' => 'Achsel Alex',
+            'role' => 'member'
+          }
+        ]
+      }
+    }.with_indifferent_access
+  end
+
+  describe '.as_json'
```
Comment (austinmoore) [inline spec/shore/client/tokens/merchant_account_spec.rb]: Is this a syntax error? Shouldn't it be `describe '.as_json' do .... end`?

File: .ruby-version (hunk 1 above)
Comment (kdelchev) [inline .ruby-version]: Set to 2.1.8 as in core.

File: README.md (hunk 3 above)
Comment (kdelchev) [inline README.md]: `shore-client` gives rubocop warning. I set it to `shore_client`.

File: lib/shore/client/tokens/access_token.rb (hunk 4 above)
Comment (kdelchev) [inline lib/shore/client/tokens/access_token.rb]: Changed to `.freeze`

File: lib/shore/client/tokens/merchant_account.rb (hunk 5 above)
Comment (kdelchev) [inline lib/shore/client/tokens/merchant_account.rb]: Moved.

File: spec/shore/client/tokens/access_token_spec.rb (hunk 6 above)
Comment (kdelchev) [inline spec/shore/client/tokens/access_token_spec.rb]: I copy-pasted the `spec_helper.rb` from messaging service and fixed all warnings.

File: spec/shore/client/tokens/merchant_account_spec.rb (hunk 7 above)
Comment (kdelchev) [inline spec/shore/client/tokens/merchant_account_spec.rb]: Fixed. There was no syntax error. Rspec allows you to mark your cases only and then add the specific test. It skip `describe` blocks and marks as pending 'it' blocks.

File: README.md (hunk 3 above)
Comment (kdelchev) [inline README.md]: Set to `shore-client` and silenced the warning.

**Pull request merged**